### PR TITLE
views: add button to use a resource

### PR DESCRIPTION
**Pull request by zannkukai**
This commit add the possibility to use a resource. If a resource can be
used, it is associated to an URL to use this resource. This commit also
add a button ('use') for each search result and in the detail view. This
button can be controlled by the 'canUse' parameter from the resource
configuration.

Co-authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

**Comment by iGormilhit** [comment]
@zannkukai Could you add a small precision on the difference between editing a resource and using it?

File: projects/rero/ng-core/src/lib/record/detail/detail.component.html [hunk 1]
```diff
@@ -17,7 +17,15 @@
 <div class="main-content">
   <ng-core-error [error]="error" *ngIf="error"></ng-core-error>
   <div class="float-right ml-4 mt-2 mb-4" *ngIf="record && adminMode.can">
-    <a href class="btn btn-sm btn-primary" [routerLink]="['../../edit', record.metadata.pid]"
+    <button *ngIf="useStatus && useStatus.can && useStatus.url"
+            id="detail-use-button"
+            class="btn btn-sm btn-outline-primary ml-1"
```
Comment (AoNoOokami) [inline projects/rero/ng-core/src/lib/record/detail/detail.component.html:22]: For the templates, I would not put this button in outline, as it is the primary action. However, the edit button should be outline in this particular case.

**Review by AoNoOokami — CHANGES_REQUESTED** [review]

File: projects/rero/ng-core/src/lib/record/detail/detail.component.html (hunk 1 above)
Comment (zannkukai) [reply projects/rero/ng-core/src/lib/record/detail/detail.component.html]: done

**Review by AoNoOokami — APPROVED** [review]

File: projects/rero/ng-core/src/lib/record/record-ui.service.ts [hunk 2]
```diff
@@ -117,27 +117,52 @@ export class RecordUiService {
...
   }
 
+  /**
+   * Redirects after saving a resource
+   * @param pid - string, pid of the record
+   * @param record - object, record to save
+   * @param recordType - string, Type of resource
+   * @param action - string, http action: create or update
+   */
+  redirectAfterSave(pid: string, record: any, recordType: string, action: string, route: ActivatedRoute) {
```
Comment (lauren-d) [inline projects/rero/ng-core/src/lib/record/record-ui.service.ts:133]: missing param for `route`

File: projects/rero/ng-core/src/lib/record/search/result/record-search-result.component.ts [hunk 3]
```diff
@@ -116,7 +129,8 @@ export class RecordSearchResultComponent implements OnInit {
    */
   constructor(
     private _componentFactoryResolver: ComponentFactoryResolver,
-    private _recordUiService: RecordUiService
+    private _recordUiService: RecordUiService,
+    private _router: Router
```
Comment (lauren-d) [inline projects/rero/ng-core/src/lib/record/search/result/record-search-result.component.ts:133]: missing param docstring

**Review by lauren-d — APPROVED** [review]

File: projects/rero/ng-core/src/lib/record/record-ui.service.ts (hunk 2 above)
Comment (zannkukai) [reply projects/rero/ng-core/src/lib/record/record-ui.service.ts:133]: done

File: projects/rero/ng-core/src/lib/record/search/result/record-search-result.component.ts (hunk 3 above)
Comment (zannkukai) [reply projects/rero/ng-core/src/lib/record/search/result/record-search-result.component.ts:133]: done

**Review by sebastiendeleze — APPROVED** [review]

**Review by iGormilhit — APPROVED** [review]
Commit message approved.

File: projects/rero/ng-core/src/lib/record/detail/detail.component.ts [hunk 4]
```diff
@@ -49,6 +49,15 @@ export class DetailComponent implements OnInit, OnDestroy {
     message: ''
   };
 
+  /**
+   * Record can be used ?
+   */
+  useStatus: ActionStatus = {
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/detail/detail.component.ts:55]: useRecord or useResource?

**Review by jma — APPROVED** [review]

File: projects/rero/ng-core/src/lib/record/detail/detail.component.ts (hunk 4 above)
Comment (zannkukai) [reply projects/rero/ng-core/src/lib/record/detail/detail.component.ts:55]: as we already use `deleteStatus` and `udpateStatus`, it makes sense to use `useStatus`. Are you agree ?
I already write a function 'useRecord()' in this component (used when you click on 'use' button.
So a solution, should be to rename all 'status' with 'resource' --> `updateResourceStatus`, `deleteResourceStatus` and  `useResourceStatus`.

**Pull request merged**